src/rust/demikernel/bindings.rs

```diff
@@ -512,10 +512,28 @@ pub extern "C" fn demi_getsockopt(
 // Standalone Functions
 //==============================================================================
 
-/// Converts a [sockaddr] into a [Ipv4Endpoint].
+/// Converts a [sockaddr] into a [SocketAddrV4].
 fn sockaddr_to_ipv4endpoint(saddr: *const sockaddr) -> Result<SocketAddrV4, Fail> {
-    // TODO: Review why we need byte ordering conversion here.
     let sin: libc::sockaddr_in = unsafe { *mem::transmute::<*const sockaddr, *const libc::sockaddr_in>(saddr) };
-    let addr: Ipv4Addr = { Ipv4Addr::from(u32::from_be_bytes(sin.sin_addr.s_addr.to_le_bytes())) };
-    Ok(SocketAddrV4::new(addr, sin.sin_port))
+    let addr: Ipv4Addr = Ipv4Addr::from(sin.sin_addr.s_addr.to_be());
+    let port: u16 = sin.sin_port.to_be();
+    Ok(SocketAddrV4::new(addr, port))
+}
+
+#[test]
+fn test_sockaddr_to_ipv4endpoint() {
+    // SocketAddrV4: 127.0.0.1:80
+    let saddr: libc::sockaddr = {
+        sockaddr {
+            sa_family: 0,
+            sa_data: [0, 80, 127, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0],
```

Comment:
For an IPv4 address, sa_family should be AF_INET (2).  Should we check this?  If we implemented sockaddr_to_SocketAddr, which enum it returns should depend upon this field.

Reply:
I understand your perspective.

However, in this PR I aimed for fixing the endianness conversion for IPv4 addresses only.

Anyways, I added TODOs on the source code to note your suggestion.